src/dhanshri/Shop.java

```diff
@@ -0,0 +1,111 @@
+package dhanshri;
+
+public class Shop {
+
+	int maggieQ, dosaQ, pouchesQ, masalaQ, panipuriQ;
+
+	void setinitialQ(int maggie, int dosa, int pouches, int panipuri, int masala) {
+		maggieQ = maggie;
+		dosaQ = dosa;
+		pouchesQ = pouches;
+		panipuriQ = panipuri;
+		masalaQ = masala;
+
+	}
+
+	void requestedQ(int reqMaggie, int reqDosa, int reqPouches, int reqPanipuri, int reqMasala,
+			boolean pAllowedstatus) {
+
+		if (reqMaggie <= maggieQ) {
+			maggieQ = maggieQ - reqMaggie;
+		} else {
+			System.out.println("sorry ,requires maggie quantity is not available we can give you " + maggieQ);
+			if (pAllowedstatus) {
+				maggieQ = 0;
```

Comment:
If request stock is not available mean we are setting direct it zero, 
what if second time your calling same method and let say you request 20 packs, then this looks not logically correct